File: 2166.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim();

const N = parseInt(input);

let valor = 0.0;

for (let i = 0; i < N; i++) {
    valor = 1 / (2 + valor);
}

valor += 1;

console.log(valor.toFixed(10));

File: 2166.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);

    double valor = 0.0;

    for (int i = 0; i < N; i++) {
        valor = 1.0 / (2.0 + valor);
    }

    valor += 1.0;

    printf("%.10lf\n", valor);
    return 0;
}
